Add feature to override the Datadog installer default package version

snowman11784 opened the pull request:
I have added a feature to override the Datadog installer default package version (mainly for Air-gapped environments).

These changes are for air-gapped environments with a custom/private registry with the installer images tagged only with a specific "version".

Currently, the ansible role does not have the ability to override the installer default package version.

**CHANGES:**

- Created a new default variable named "**ansible_installer_version**" to set the installer version explicitly.
- Added the environment variable "**DD_INSTALLER_DEFAULT_PKG_VERSION_DATADOG_INSTALLER**" to the "Bootstrap the installer" task and set it to "ansible_installer_version".
- Added version to the "installer_registry_config"
- Updated README and usage examples under the "Air-gapped environments" section
- Added CircleCI test to validate the installer version and installation of the Datadog installer packages for air-gapped environments

For more background information, see #651

Commit:
Added feature to override the Datadog installer default package version (mainly for Air-gapped environments)

Commit:
Added CI test for custom air-gapped environment

Commit:
Added version to installer_registry_config
Removed quotes from datadog_agent_version and datadog_installer_version

Commit:
Added CircleCI tests for test_installer_air_gapped
Removed the quotes in the datadog_installer_version value from README
example

Commit:
Added remote_updates to the CircleCI test_installer_air_gapped test

janine-c commented on .circleci/config.yml:
```diff
@@ -405,7 +457,7 @@ workflows:
           matrix:
             parameters:
-              ansible_version: ["2_10", "3_4", "4_10", "9_4"]
+              ensible_version: ["2_10", "3_4", "4_10", "9_4"]
```
Want to double check - is this a typo?

snowman11784 replied:
@janine-c yes it was a typo. My cursor was even on the wrong test.
In the typo commit, I also added installer_version to matrix test parameters for "test_installer_air_gapped".

Commit:
Fix typo from ensible to ansible in one of the CircleCI test_install jobs
Added test parameter "installer_version" for test_installer_air_gapped

Commit:
Renamed CircleCI test for air-gapped to reference testing only
RHEL-based OS
Added more testing parameters for yum_repo, yum_repo_config_enabled,
installer_registry, and dd_site

Commit:
Removed test parameters since I went over jobs limit in previous commit

Commit:
Updated yum_repo path to reference version 7 stable x86_64 repository

Commit:
Removed yum_repo_config_enabled false test parameter

Commit:
Updated default vars datadog_yum_repo path

swang392 commented on .circleci/config.yml:
```diff
@@ -331,6 +331,73 @@ jobs:
+      yum_repo:
+        type: string
+      yum_repo_config_enabled:
+        type: string
```
```suggestion
```

swang392 commented on .circleci/config.yml:
```diff
@@ -331,6 +331,73 @@ jobs:
+          -e datadog_site="<<parameters.dd_site>>"
+          -e datadog_installer_registry="<<parameters.installer_registry>>"
+          -e datadog_yum_repo="<<parameters.yum_repo>>"
+          -e datadog_yum_repo_config_enabled="<<parameters.yum_repo_config_enabled>>"
```
```suggestion
```

swang392 commented on .circleci/config.yml:
```diff
@@ -458,6 +525,19 @@ workflows:
+              dd_site: ["datadoghq.com", "ddog-gov.com"]
+              installer_registry: ["install.datadoghq.com"]
+              yum_repo: ["https://yum.datadoghq.com/stable/7/x86_64"]
+              yum_repo_config_enabled: ["true"]
```
```suggestion
```

swang392 commented on ci_test/install_installer_air_gapped.yaml:
```diff
@@ -0,0 +1,15 @@
+    datadog_enabled: false
+    datadog_skip_running_check: true
+    datadog_installer_registry: "install.datadoghq.com"
+    datadog_installer_version: 7.63
```
```suggestion
    datadog_installer_version: 7.62
```
Right now the default version is also 7.63, so setting it to an older version can help verify that its installing the `datadog_installer_version` rather than latest. with 7.62 here the installer version is `7.62.1-installer-0.11.0`

swang392 commented:
Thank you for the contribution!! In my comments above I've simplified the test matrix, since right now ci only runs [ci/circleci: /yum.datadoghq.com/stable/7/x86_64-true](https://circleci.com/gh/DataDog/ansible-datadog/146517) instead of all of the integration test combinations.

snowman11784 commented:
> Thank you for the contribution!! In my comments above I've simplified the test matrix, since right now ci only runs [ci/circleci: /yum.datadoghq.com/stable/7/x86_64-true](https://circleci.com/gh/DataDog/ansible-datadog/146517) instead of all of the integration test combinations.

@swang392 my changes were specific to RHEL, I should also include updates to install installer for Red Hat, SuSE, and Debian.

swang392 commented:
@snowman11784 I tested on suse and debian and it worked as expected.

snowman11784 committed:
Apply suggestions from code review

Co-authored-by: Sarah Wang <sarah.wang@datadoghq.com>
